### PR TITLE
chat-archiver: Download media that is linked in chat

### LINKED ISSUE #403: chat-archiver: Save copies of images posted
Image links tend to rot quickly. To archive the chat we should also archive any images linked.

### PR DESCRIPTION
**Pull request by ekimekim**
Most of the logic here comes down to three thorny issues:
- How to identify links in chat
- Safely fetching from potentially-malicious URLs
- How to store that in the filesystem

We store things in such a way that it works with backfiller, so you can copy any data you miss
from other instances.

The purpose of all this is to provide an archive of images etc linked in chat *as they were at the time*, as links can rot quickly.
Special care is taken to re-check the content each time it's linked in case it returns a different result.

This may end up incurring a lot of disk usage. The expectation is that we'll only enable this for the main nodes.

Closes #403

### CODE REVIEW

**Comment by ekimekim** [comment]
I ran this URL parser over last year's chat logs and it seems to work well. I compared it to a much looser version with a lot of false positives and it didn't have any false negatives (things it failed to find) while eliminating the vast majority of the false positives.

**Comment by ekimekim** [comment]
Added a second part, also fixes #405

File: common/common/media.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+	content_types=("image", "video", "application/pdf"),
+	max_redirects=5,
+	retries=3,
+	retry_interval=1,
+	chunk_size=64*1024, # 64KiB
+):
+	"""Make a GET request to a potentially malicious URL and download the content to file.
+	We check the following:
+	- That the host is a public IP
+	- That the response does not exceed given max size (default 100MB)
```
Comment (chrusher) [inline common/common/media.py:79]: The max size is set to 128 MB by default not 100 MB

File: docker-compose.jsonnet [hunk 2]
```diff
@@ -21,7 +21,7 @@
     restreamer: true,
     backfiller: true,
     cutter: true,
-    sheetsync: false,
+    sheetsync: true,
```
Comment (chrusher) [inline docker-compose.jsonnet:24]: Should sheetsync be active by default?

**Review by chrusher — APPROVED** [review]
Looks good to me apart from a couple minor queries

File: common/common/media.py (hunk 1 above)
Comment (ekimekim) [reply common/common/media.py:79]: ah good catch, that number changed a few times between writing the docs and the final thing.

File: docker-compose.jsonnet (hunk 2 above)
Comment (ekimekim) [reply docker-compose.jsonnet:24]: whoops, this was testing code that leaked in

**Comment by ekimekim** [comment]
I'm just going to merge this without fully testing it, a decision I could not possibly come to regret

**Pull request merged**